### DISCUSSION TITLE
User roles per client/tenant?

### DISCUSSION THREAD

**Discussion by jhermsen** [simpleidserver/SimpleIdServer#772] [Q&A]
There are so many options that I might have missed it, but I was wondering if it is possible to:
- have a user with access to multiple clients, but with different roles?
- have a user with access to a single client, but that client has multiple tenants(different hostnames), and different roles?

Is this what realms can do or is there any other option currently available for this?

**Comment by simpleidserver** [comment]
Hello,

**User with Access to Multiple Clients with Different Roles**

This scenario is supported by SimpleIdServer. You can assign a client role to a user by following these steps:

* Navigate to a client, select the Roles tab, and add a new client role like admin.
* Navigate to a group, select the Roles tab, and assign the new client role.
* Navigate to a user, select the Groups tab, and assign the previously created group.

**User with Access to a Single Client with Multiple Tenants and Different Roles**

Currently, realms are completely separated; therefore, a client present in one realm cannot be assigned to another realm. Although it is possible to do this without any changes in the database, we can update our base code to support this scenario.

Best regards,

SID

**Reply by jhermsen** [reply]
Thanks again for the quick response.
